pkgs/tools/misc/ili2c/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub, jdk, ant, makeWrapper, jre }:
+
+stdenv.mkDerivation rec {
+  name = "ili2c-${version}";
+  version = "4.7.11";
+
+  buildInputs = [ ant jdk makeWrapper ];
```

Comment:
Does `ant` belong in `nativeBuildInputs`?

Reply:
Probably. :thinking:
`nativeBuildInputs` is for buildtime-only dependencies (not needed at runtime), isn't it? (And `buildInputs` for dependencies needed at both, build-time and runtime?)

And `jdk` and `makeWrapper` should probably be buildtime-only, too, shouldn't they ? (I think only `jre` should be a runtime dependency, in this case. And as that should be implicit due to the wrapper script's closure, I probably won't even have to specify it as such?)

Though https://nixos.org/nixpkgs/manual/#sec-language-java currently suggests `buildInputs` for `jdk`, `ant` and `makeWrapper`. Should the manual be revised in that regard?

Reply:
I've now put all the dependees (except for JRE) into `nativeBuildInputs`: https://github.com/NixOS/nixpkgs/compare/b6388f7be50dbb30e56b18acb727fcabd0379fb5..2aaa8876ff68e454ff60fc9a2b6d2174df6bb17a#diff-efa52252ac746605798baf0ce1e25f13R7

Reply:
@das-g Everything you said is correct. I also assume there was a mistake in the manual. There aren't many `java` based projects packaged for NixOS which actually build from source, so I assume mistakes were made and copy/pasted around.

Reply:
I'm also under the impression you need a `propagatedBuildInputs = [ jre ];`

Reply:
> I'm also under the impression you should have a `propagatedBuildInputs = [ jre ];`

Do I? The `ili2c` command seems to run fine without specifying that (and I don't have any JRE installed system-wide). Or would that be needed if other software wants to use ili2c as a library? (Which might happen, as AFAIK [ili2db](https://www.interlis.ch/downloads/ili2db) depends on ili2c. Though then I should probably put the ili2c JAR in `$out/share/java` instead of a "private" (package-specific) subfolder?)

Reply:
> I also assume there was a mistake in the manual. There aren't many `java` based projects packaged for NixOS which actually build from source, so I assume mistakes were made and copy/pasted around.

Let's fix the documentation then: #64879 :smiley:

Reply:
@das-g Awesome work!